packages/core/src/Nodes/NodeDefinition.ts

```diff
@@ -0,0 +1,138 @@
+import { NodeCategory } from './Registry/NodeCategory';
+
+export type SocketsDefinition = Record<string, string>;
```

Comment:
As mentioned in a comment, this should allow a more complex configuration like:
```suggestion

type SocketsDefinitionExtended = {
  valueType: string;
  defaultValue?: unknown;
  choices?: unknown[];
}

export type SocketsDefinition = Record<string, string | SocketsDefinitionExtended>;
```

This would also allow us to extend it to add more props like helpDescription or label to provide a better user experience on the visual editor